File: source/StockCalculator/Rate.m
//
//  Rate.m
//  StockCalculator
//
//  Created by zuohaitao on 15/10/28.
//  Copyright © 2015年 zuohaitao. All rights reserved.
//

#import "Rate.h"

@implementation RateOld

-(instancetype)init{
    self = [super init];
    return self;
}

-(float)commission {
    return [[NSUserDefaults standardUserDefaults] floatForKey:@"commission"];
}


-(float) stamp {
    return [[NSUserDefaults standardUserDefaults] floatForKey:@"stamp"];
}


-(float)transfer {
    return [[NSUserDefaults standardUserDefaults] floatForKey:@"transfer"];
}

-(void)setCommission:(float)c {
    [[NSUserDefaults standardUserDefaults] setFloat:c forKey:@"commission"];
}

-(void)setStamp:(float)s {
    [[NSUserDefaults standardUserDefaults] setFloat:s forKey:@"stamp"];
}

-(void)setTransfer:(float)t {
    [[NSUserDefaults standardUserDefaults] setFloat:t forKey:@"transfer"];
}



@end

File: source/StockCalculator/Rate.swift
//
//  Rate.swift
//  StockCalculator
//
//  Created by zuohaitao on 15/11/20.
//  Copyright © 2015年 zuohaitao. All rights reserved.
//

import Foundation

@objc(Rate)
class Rate:NSObject {
    override func valueForKey(key: String) -> AnyObject? {
        return NSUserDefaults.standardUserDefaults().doubleForKey(key);
    }
    override func setValue(value: AnyObject?, forKey key: String) {
        NSUserDefaults.standardUserDefaults().setDouble((value as! Double), forKey: key)
    }
    var stamp:Double {
        get {
            let v : Double! = self.valueForKey(__FUNCTION__) as? Double
            if v == nil {
                return 0.00
            }
            return v
        }

    }
    var transfer:Double {
        get {
            let v : Double! = self.valueForKey(__FUNCTION__) as? Double
            if v == nil {
                return 0.00
            }
            return v
        }
    }
    var commission:Double {
        get {
            let v : Double! = self.valueForKey(__FUNCTION__) as? Double
            if v == nil {
                return 0.00
            }
            return v
        }
    }
}